code/game/machinery/turret_control.dm

```diff
@@ -180,9 +180,32 @@
 	TC.ailock = ailock
 
 	if(istype(control_area))
+		var/lethal_only = 0 // To check if there is a lethal turret in the area.
+		var/both_mode = 0
 		for (var/obj/machinery/porta_turret/aTurret in control_area)
+			// If turret only has lethal mode - lock switching modes
+			if(aTurret.egun == 0)
+				lethal_only = 1
+				egun = 0
```

Comment:
This is a bad place to modify this, IMO. It'd mean potentially removing an option whenever the user inputs values. Maybe make porta_turret/Initialize call a function on the resident zone's turret controls so that they update their egun values?

Basically: this proc should only set the states of turrets, not validate said states.

Reply:
@skull132 User cannot input values in turret because turret controls are locked once turret control panel is in the area. Other thing is that Initialize is only being called once the panel is spawned. So if I will put it into initialize, it will only update these settings once you spawn control panel, meaning if someone made a new turret it will not properly scan it, and set it up. So if want this out, then we need other proc, which will essentially have same function, just be as seperate proc.

Reply:
> User cannot input values in turret because turret controls are locked once turret control panel is in the area.

wat

> Other thing is that Initialize is only being called once the panel is spawned.

Yes, that is why I asked you put into porta_turret/Initialize as well. And a final set of calls into the Destroy of both. A standard update chain, basically.

Reply:
Oh, I see the idea. Sorry, I misunderstood you.
Making `porta_turret/Initialize` to call a function turret controls will do.